gamemodes/terrortown/gamemode/weaponry_shd.lua

```diff
@@ -124,6 +124,53 @@ function WEPS.ResetRoleWeaponCache()
     end
 end
 
+local function PlayerOwnsWepOrItem(ply, classOrId)
+    if isstring(classOrId) then
+        for _, wep in ipairs(ply:GetWeapons()) do
+            if wep:GetClass() == classOrId then
```

Comment:
This utility method is a safer way to get the weapon class. There are a few cases I've seen in the Randomat where this works and `:GetClass` fails
```suggestion
            if WEPS.GetClass(wep) == classOrId then
```